InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Exceptions.Modify.cs

```diff
@@ -0,0 +1,235 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using InternTrack.Core.Api.Models.Interns.Exceptions;
+using InternTrack.Core.Api.Models.Interns;
+using Microsoft.Data.SqlClient;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+using FluentAssertions;
+using Microsoft.EntityFrameworkCore;
+
+namespace InternTrack.Core.Api.Tests.Unit.Services.Foundations.Interns
+{
+    public partial class InternServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowCriticalDepdnencyExceptionOnModifyIfSqlErrorOccursAndLogItAsync()
+        {
+            // given
+            DateTimeOffset datetime = GetRandomDateTime();
+            Intern randomIntern = CreateRandomIntern(datetime);
+            randomIntern.CreatedDate = datetime;
+            randomIntern.UpdatedDate = randomIntern.CreatedDate.AddMinutes(GetRandomNumber());
+
+            SqlException sqlException = GetSqlException();
+
+            var failedInternStorageException =
+                new FailedInternStorageException(sqlException);
+
+            var expectedInternDependencyExcetpion =
+                new InternDependencyException(failedInternStorageException);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffset())
+                    .Returns(randomIntern.UpdatedDate);
```

Comment:
Instead of having to build a perfect object just so we can throw an exception. Please attach the exception the `dateTimeBroker`